File: HW3/hw3_1.py
from pyomo.environ import *

# ============================================================
#                           Model
# ============================================================
model = ConcreteModel()

model.Qs = Var(domain=NonNegativeReals)
model.Qw = Var(domain=NonNegativeReals)
model.R = Var(range(1, 4), domain=NonNegativeReals)

model.objective = Objective(expr=model.Qs + model.Qw, sense=minimize)

model.constraint1 = Constraint(expr = model.Qs - model.R[1] == 30)
model.constraint2 = Constraint(expr = model.R[1] - model.R[2] == 30)
model.constraint3 = Constraint(expr = model.R[2] - model.R[3] == -123)
model.constraint4 = Constraint(expr = model.R[3] - model.Qw == -102)

# ============================================================
#                           Solver
# ============================================================
solver = SolverFactory('gurobi')
results = solver.solve(model, tee=True)

# ============================================================
#                           Output
# ============================================================
print(f"Optimal Qh: {value(model.Qs):.2f} MW")
print(f"Optimal Qc: {value(model.Qw):.2f} MW")
for i in range(1, 4):
    print(f"R[{i}]: {value(model.R[i]):.2f} MW")

File: HW3/hw3_1.jl
using JuMP
using Gurobi

# ============================================================
#                           Model
# ============================================================
model = Model(Gurobi.Optimizer)

@variable(model, Qs >= 0)
@variable(model, Qw >= 0)
@variable(model, R[1:3] >= 0)

@objective(model, Min, Qs + Qw)

@constraint(model, Qs - R[1] == 30)
@constraint(model, R[1] - R[2] == 30)
@constraint(model, R[2] - R[3] == -123)
@constraint(model, R[3] - Qw == -102)

# ============================================================
#                           Solver
# ============================================================
optimize!(model)

# ============================================================
#                           Output
# ============================================================
println("Optimal Qs: ", value(Qs), "MW")
println("Optimal Qw: ", value(Qw), "MW")
for i in 1:3
    println("R",i, " value: ", value(R[i]), "MW")
end